label_studio/ml_model_providers/models.py

```diff
@@ -42,3 +43,20 @@ class ModelProviderConnectionScopes(models.TextChoices):
 
     def has_permission(self, user):
         return user.active_organization == self.organization
+
+    def validate_api_key(self):
+        """
+        Checks if API key provided is valid
+
+        Return: boolean if API key is valid or not
+        """
+        if self.provider == self.ModelProviders.OPENAI:
+            try:
+                print("creating open ai client")
+                client = OpenAI(api_key=self.api_key)
+                print("successfully created open ai client")
+                client.models.list()
+            except AuthenticationError as e:
+                return False
+            else:
+                return True
```

Comment:
As above, I think it might be more helpful to let this AuthenticationError bubble up, or re-raise it. then in the serializer you can catch this exception and re-raise as a `ValidationError`